### PR TITLE
more type stuff

### PR DESCRIPTION
**Pull request by Bike**
mainly type inference for aref, aset, un/box

### CODE REVIEW

File: Code/Cleavir/Type-inference/transfer.lisp [hunk 1]
```diff
@@ -94,44 +94,76 @@
 		input-bag))))
 
 (defmethod one-successor-transfer
-    ((instruction cleavir-ir:short-float-unbox-instruction) input-bag)
-  (let ((output (first (cleavir-ir:outputs instruction))))
-    (update output 'unboxed-short-float input-bag)))
+    ((instruction cleavir-ir:aref-instruction) input-bag)
+  ;; TODO: update array
+  (let ((index (second (cleavir-ir:inputs instruction)))
+	(output (first (cleavir-ir:outputs instruction)))
```
Comment (Harleqin) [inline Code/Cleavir/Type-inference/transfer.lisp]: This looks like the indentation is a bit off. Artifact of the github diff viewer or real?

**Comment by robert-strandh** [comment]
Must be an artifact.  It looks good in Emacs here.

File: Code/Cleavir/Type-inference/transfer.lisp (hunk 1 above)
Comment (larsbrinkhoff) [inline Code/Cleavir/Type-inference/transfer.lisp]: The github diff viewer handles TABs just as badly as diff output to a terminal.

Comment (Harleqin) [inline Code/Cleavir/Type-inference/transfer.lisp]: I do not think that TABs are a good idea, especially in Lisp code, where standard indentation is usually not to some arbitrary level, but to a specific column.

Comment (Bike) [inline Code/Cleavir/Type-inference/transfer.lisp]: i just write everything in slime. i don't know if it uses tab characters or what, though i would think no.

Comment (larsbrinkhoff) [inline Code/Cleavir/Type-inference/transfer.lisp]: I believe Emacs by default indents using tabs at the beginning of the line, and then spaces to reach the desired column.

Comment (Harleqin) [inline Code/Cleavir/Type-inference/transfer.lisp]: I recommend (setq-default indent-tabs-mode nil) in every Emacs init file (.emacs or .emacs.d/init.el).

Comment (Bike) [inline Code/Cleavir/Type-inference/transfer.lisp]: done

**Pull request merged**